bitcoinetl/service/btc_block_range_service.py

```diff
@@ -32,9 +32,12 @@ def __init__(self, bitcoin_rpc):
         graph = BlockTimestampGraph(bitcoin_rpc)
         self._graph_operations = GraphOperations(graph)
 
-    def get_block_range_for_date(self, date):
-        start_datetime = datetime.combine(date, datetime.min.time().replace(tzinfo=timezone.utc))
-        end_datetime = datetime.combine(date, datetime.max.time().replace(tzinfo=timezone.utc))
+    def get_block_range_for_date(self, date, start_hour=0, end_hour=24):
+        max_datetime = datetime(date.year, date.month, date.day, 23, 59, 59)
+        start_datetime = date + timedelta(hours=start_hour)
+        end_datetime = date + timedelta(hours=end_hour)
```

Comment:
For consistency start_hour and end_hour should be both inclusive. All apis in blockchain-etl use inclusive intervals for block numbers and the time (usually up to a second resolution) as it's more intuitive for discrete values and it follows the Ethereum API conventions (e.g. in getLogs https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getlogs startBlock and endBlock are both inclusive). 

Could you change the calculations here so that end_datetime would be the last second of end_hour? And the default value for end_hour to be 23.

Reply:
Thanks for the advice, now fixed.